src/OpenTelemetry.Api/Trace/SpanHelper.cs

```diff
@@ -0,0 +1,93 @@
+﻿// <copyright file="SpanHelper.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System.Collections.Concurrent;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// A collection of helper methods to be used when building spans.
+    /// </summary>
+    public static class SpanHelper
+    {
+        private static readonly ConcurrentDictionary<StatusCanonicalCode, string> CanonicalCodeToStringCache = new ConcurrentDictionary<StatusCanonicalCode, string>();
```

Comment:
Suggestion for a separate PR depending if Status doesn't change much: since there are only 17 canonical codes an array should suffice.

Reply:
Only 17 he says :) I updated the PR since I was in there trying to work on some of your other feedback.